interceptors/logging/interceptors_test.go

```diff
@@ -88,47 +85,50 @@ func (l LogLines) Swap(i, j int) {
 	l[i], l[j] = l[j], l[i]
 }
 
-type baseMockLogger struct {
-	// Shared. It's slice on purpose to find potential duplicates.
+type sharedResults struct {
+	// Contains a shared slice among all mockLoggers,
+	// for appending the Log results
 	lines []LogLine
 	m     sync.Mutex
 }
 
-func (l *baseMockLogger) Add(line LogLine) {
+func (l *sharedResults) Lines() []LogLine {
 	l.m.Lock()
 	defer l.m.Unlock()
 
-	l.lines = append(l.lines, line)
-}
+	retLines := make([]LogLine, len(l.lines))
+	copy(retLines, l.lines)
 
-func (l *baseMockLogger) Lines() []LogLine {
-	l.m.Lock()
-	defer l.m.Unlock()
-
-	return l.lines
+	return retLines
 }
 
 type mockLogger struct {
-	*baseMockLogger
+	*sharedResults
```

Comment:
Great, thanks for confirming what I understood! 

> So we can either sort lines before comparing or remove sharedResult and keep references to all create loggers with With? Whatever works. Maybe if you name it output it will make more sense? (:

I think, I would keep the original method of comparing the results, that sounds much more dependable. The references might create a bug-prone experience for comparing the results, so I would stick with the original method.

I have renamed the `sharedResults` into `mockStdOutput`, and have added some comments for how we are using it as a shared struct. Would be happy to add more comments if needed. 

I have also sorted the output lines so that we have a definitive order for comparing the values.